code/datums/crew_objective.dm

```diff
@@ -307,11 +330,143 @@ ABSTRACT_TYPE(/datum/objective/crew/janitor)
+	shotgun
+		explanation_text = "Don't lose your shotgun!"
+		check_completion()
+			if(owner.current && owner.current.check_contents_for(/obj/item/gun/kinetic/riotgun))
+				return 1
+			else
+				return 0
+	drinks
+		var/completed = 0
+		var/ids[3]
+		New()
+			..()
+			var/list/cocktails = concrete_typesof(/datum/reagent/fooddrink/alcoholic)
+			var/list/blacklist = list(
+				/datum/reagent/fooddrink/alcoholic/bitters,
+				/datum/reagent/fooddrink/alcoholic/beer,
+				/datum/reagent/fooddrink/alcoholic/bojack,
+				/datum/reagent/fooddrink/alcoholic/bourbon,
+				/datum/reagent/fooddrink/alcoholic/champagne,
+				/datum/reagent/fooddrink/alcoholic/cider,
+				/datum/reagent/fooddrink/alcoholic/cocktail_grog,
+				/datum/reagent/fooddrink/alcoholic/curacao,
+				/datum/reagent/fooddrink/alcoholic/dbreath,
+				/datum/reagent/fooddrink/alcoholic/freeze,
+				/datum/reagent/fooddrink/alcoholic/gin,
+				/datum/reagent/fooddrink/alcoholic/mead,
+				/datum/reagent/fooddrink/alcoholic/moonshine,
+				/datum/reagent/fooddrink/alcoholic/ricewine,
+				/datum/reagent/fooddrink/alcoholic/rum,
+				/datum/reagent/fooddrink/alcoholic/tequila,
+				/datum/reagent/fooddrink/alcoholic/vermouth,
+				/datum/reagent/fooddrink/alcoholic/vodka,
+				/datum/reagent/fooddrink/alcoholic/wine,
+				/datum/reagent/fooddrink/alcoholic/wine/white
+			)
+			cocktails -= blacklist
+			var/list/names[3]
+			for(var/i = 1, i <= 3, i++)
+				var/choiceType = pick(cocktails)
+				var/datum/reagent/fooddrink/instance =  new choiceType
+				names[i] = instance.name
+				ids[i] = instance.id
+			explanation_text = "Mix a [names[1]], [names[2]], and [names[3]] using your cocktail shaker."
+		check_completion()
+			return completed == DRINK_OBJ_DONE //Uses bit flags
+
+ABSTRACT_TYPE(/datum/objective/crew/chef)
+/datum/objective/crew/chef
+	var/list/blacklist = list(
+		/obj/item/reagent_containers/food/snacks/burger/humanburger,
+		/obj/item/reagent_containers/food/snacks/donut/custom/robust,
+		/obj/item/reagent_containers/food/snacks/ingredient/meat/humanmeat,
+		/obj/item/reagent_containers/food/snacks/ingredient/meat/mysterymeat/nugget/flock,
+		/obj/item/reagent_containers/food/snacks/ingredient/pepperoni,
+		/obj/item/reagent_containers/food/snacks/meatball,
+		/obj/item/reagent_containers/food/snacks/mushroom,
+		/obj/item/reagent_containers/food/snacks/pickle/trash,
+		/obj/item/reagent_containers/food/snacks/pizza/xmas,
+		/obj/item/reagent_containers/food/snacks/plant,
+		/obj/item/reagent_containers/food/snacks/plant/glowfruit/spawnable,
+	)
+	cake
+		var/choices[3]
+		var/completed = 0
+		New()
+			..()
+			//Result is cached so it should just access the list after the first time
+			var/list/ingredients = concrete_typesof(/obj/item/reagent_containers/food/snacks)
+			ingredients -= src.blacklist
+			ingredients -= concrete_typesof(/obj/item/reagent_containers/food/snacks/ingredient/egg/critter)
+			var/list/names[3]
+			for(var/i = 1, i <= 3, i++)
```

Comment:
Same issue with 3, use a define so `choices`, `names` and iterating over them are coupled.

Reply:
Done